exarl/workflows/workflow_vault/sync_learner.py

```diff
@@ -588,6 +598,12 @@ def episode_round(self, exalearner):
             if nepisodes < ExaComm.agent_comm.size - ExaComm.num_learners:
                 nepisodes = ExaComm.agent_comm.size - ExaComm.num_learners
 
+        # Add episode and step counters to env
+        if not hasattr(exalearner.env.env, "workflow_episode"):
+            setattr(exalearner.env.env, "workflow_episode", 0)
```

Comment:
Can this be set in the base class? Also, I would consider renaming this to something like `env_attrib` to avoid confusion.

Reply:
Doesn't seem to work with the latest change.  The log is pasted.  Notice the episode doesn't change.  The steps under the file will always count up and never resets so don't be distracted by it.  The problem is the zeros in the second column.

I ran bandit noise.

`steps,episode,total_return,episode_len,episode_return,total_regret
1,0,0.45790139033734933,1,0.45790139033734933,0.29999999999999993
2,0,1.4482423366368231,1,0.9903409462994738,0.29999999999999993
3,0,1.7945463326263567,1,0.34630399598953343,0.9999999999999999
4,0,2.1400488920633416,1,0.3455025594369851,1.6999999999999997
5,0,2.968004016341189,1,0.8279551242778473,1.8999999999999997
6,0,3.3470805698391075,1,0.3790765534979185,2.3999999999999995
7,0,3.4050097309709657,1,0.05792916113185799,3.2999999999999994
8,0,4.244400409964991,1,0.8393906789940251,3.499999999999999
9,0,4.78725559824944,1,0.5428551882844498,3.899999999999999
10,0,5.744342890906492,1,0.9570872926570517,4.099999999999999`

Reply:
Pushed a fix. Here's the log from running bandit_noise:
```
steps,episode,total_return,episode_len,episode_return,total_regret
1,0,0.28883076318295275,1,0.28883076318295275,0.7
2,1,0.27105079241379687,1,-0.01777997076915588,1.7
3,2,0.5368999476079545,1,0.2658491551941576,2.5
4,3,1.157450964660932,1,0.6205510170529774,2.9
5,4,1.44681092480571,1,0.289359960144778,3.5999999999999996
6,5,1.8153334542082853,1,0.36852252940257535,4.3
7,6,2.004351916667897,1,0.18901846245961146,5.1
8,7,2.511834186336497,1,0.5074822696685998,5.6
9,8,2.6892486712768346,1,0.17741448494033785,6.3999999999999995
10,9,3.6157428362352624,1,0.9264941649584275,6.3999999999999995
```

Reply:
So I agree the new change to the api works.  I think it is doing something similar to what I had in the sense that it is adding workflow_episodes to the env just that it is doing it in the wrapper instead of the workflow.  Before going this route, I would want to see a good reason to keep the wrapper.  This solution still doesn't let you access anything stored in the environment.  Do we think this is enough of a reason to keep this wrapper?  Are there other cases we would reuse a similar pattern?

Reply:
Gym wrappers allows one to [extend the functionality](https://hub.packtpub.com/openai-gym-environments-wrappers-and-monitors-tutorial/) of the environment and one can modify the observations, actions, and rewards of existing environments. That's one of the reasons I would like to keep the env base class but should think about adding more functionality to it.